docker-host/main.go

```diff
@@ -0,0 +1,164 @@
+
+	if dockerHost != cli.DaemonHost() {
+		log.Fatalf("Could not connect to %s", dockerHost)
+	}
+	c := cron.New()
+	pruneImages(cli, c)
+	removeExited(cli, c)
+	updateImages(cli, c)
+	c.Start()
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		log.Fatalf("could not run command: ", err)
+	}
+}
+
+func pruneImages(client *client.Client, c *cron.Cron) {
+	c.AddFunc(pruneImagesSchedule, func() {
+		log.Println("Starting image prune")
+		ageFilter := filters.NewArgs()
+		pruneDanglingFilter := filters.NewArgs()
+		ageFilter.Add("until", pruneImagesUntil)
+		pruneDanglingFilter.Add("dangling", danglingFilter)
+
+		// # prune all images older than 7 days or what is specified in the environment variable
+		_, err := client.ImagesPrune(context.Background(), ageFilter)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		// # prune all docker build cache images older than 7 days or what is specified in the environment variable
+		_, buildErr := client.BuildCachePrune(context.Background(), types.BuildCachePruneOptions{Filters: ageFilter})
+		if buildErr != nil {
+			log.Println(buildErr)
+		}
+		// # after old images are pruned, clean up dangling images
+		_, pruneErr := client.ImagesPrune(context.Background(), pruneDanglingFilter)
+		if pruneErr != nil {
+			log.Println(pruneErr)
+		}
+		log.Println("Prune complete")
+	})
+}
+
+func removeExited(client *client.Client, c *cron.Cron) {
+	c.AddFunc(removeExitedSchedule, func() {
+		log.Println("Starting removeExited")
+		ctx := context.Background()
+		statusFilter := filters.NewArgs()
+		statusFilter.Add("status", "exited")
+		containers, err := client.ContainerList(ctx, types.ContainerListOptions{
+			Filters: statusFilter,
+		})
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		// # remove all exited containers
+		for _, container := range containers {
+			err := client.ContainerRemove(ctx, container.ID, types.ContainerRemoveOptions{
+				Force:         true,
+				RemoveVolumes: true,
+			})
+			if err != nil {
+				log.Println(err)
+			}
+		}
+		log.Println("removeExited complete")
+	})
+}
+
+func updateImages(client *client.Client, c *cron.Cron) {
+	c.AddFunc(updateImagesSchedule, func() {
+		log.Println("Starting update images")
+		ctx := context.Background()
+		filters := addFilters(repositoriesToUpdate)
+		images, err := client.ImageList(ctx, types.ImageListOptions{Filters: filters})
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		var imgRepoTags []string
+		for _, img := range images {
+			imgRepoTags = append(imgRepoTags, img.RepoTags...)
+		}
+
+		// # Iterates through all images that have the name of the repository we are interested in in it
+		for _, image := range imgRepoTags {
+			out, err := client.ImagePull(ctx, image, types.ImagePullOptions{})
+			log.Println("Image to update", image)
+
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+			defer out.Close()
+			_, error := io.Copy(os.Stdout, out)
```

Comment:
I think here you can just use `io.Discard` instead of `os.Stdout` to achieve "quiet" builds. It seems to be pretty much what the docker CLI does: https://github.com/docker/cli/blob/master/cli/command/image/trust.go#L283

Reply:
Yep this works

Logs have gone from:
```
2023/08/14 07:00:00 Starting update images
2023/08/14 07:00:03 Image to update uselagoon/nginx-drupal:latest
{"status":"Pulling from uselagoon/nginx-drupal","id":"latest"}
{"status":"Digest: sha256:596d5c28eb96efec6f0e48a8e249b6ff7f9786c1002f05c4521e4ae96aa5fde5"}
{"status":"Status: Image is up to date for uselagoon/nginx-drupal:latest"}
2023/08/14 07:00:08 Image to update uselagoon/node-18:latest
{"status":"Pulling from uselagoon/node-18","id":"latest"}
{"status":"Digest: sha256:5bc3bf621ed3cdc2fd7fb483162ea04bcec50ce7127b9dd1a6d3f55a260e8821"}
{"status":"Status: Image is up to date for uselagoon/node-18:latest"}
2023/08/14 07:00:10 Image to update amazeeio/dockerize:latest
{"status":"Pulling from amazeeio/dockerize","id":"latest"}
{"status":"Digest: sha256:1f5c9334d0be68b8277eb5eddf6f87a5969cbd2c6857dc0586e1f926ea99e176"}
{"status":"Status: Image is up to date for amazeeio/dockerize:latest"}
2023/08/14 07:00:10 Update images complete
```

To:
```
2023/08/14 23:15:00 Starting update images
2023/08/14 23:15:02 Image to update uselagoon/nginx-drupal:latest
2023/08/14 23:15:05 Image to update uselagoon/node-18:latest
2023/08/14 23:15:07 Image to update amazeeio/dockerize:latest
2023/08/14 23:15:07 Update images complete
```